moveit_simple/src/moveit_simple.cpp

```diff
@@ -775,6 +775,44 @@ bool Robot::getIK(const Eigen::Affine3d pose, std::vector<double> & joint_point,
 
 
 
+bool Robot::isNearSingular(const std::vector<double> & joint_point) const
+{
+  std::lock_guard<std::recursive_mutex> guard(m_);
+  double threshold = 1e-10; // How to determine this threshold?
```

Comment:
This could also be a parameter you pass in, but I think the user will be more confused by this than we are : /